pkg/network/dhcp/dhcp_configurator.go

```diff
@@ -29,3 +54,26 @@ func (d Configurator) ImportConfiguration(ifaceName string) (*cache.DhcpConfig,
 func (d Configurator) ExportConfiguration(config cache.DhcpConfig) error {
 	return d.cacheFactory.CacheDhcpConfigForPid(d.launcherPID).Write(config.Name, &config)
 }
+
+func (d Configurator) EnsureDhcpServerStarted(podInterfaceName string, dhcpConfig cache.DhcpConfig, dhcpOptions *v1.DHCPOptions) error {
+	dhcpStartedFile := fmt.Sprintf("/var/run/kubevirt-private/dhcp_started-%s", podInterfaceName)
+	_, err := os.Stat(dhcpStartedFile)
+	if os.IsNotExist(err) {
+		if err := d.startServer(dhcpConfig, d.advertisingIfaceName, dhcpOptions); err != nil {
+			return fmt.Errorf("failed to start DHCP server for interface %s", podInterfaceName)
+		}
+		newFile, err := os.Create(dhcpStartedFile)
+		if err != nil {
+			return fmt.Errorf("failed to create dhcp started file %s: %s", dhcpStartedFile, err)
+		}
+		newFile.Close()
+	}
+	return nil
+}
+
+func (d Configurator) startServer(config cache.DhcpConfig, advertisingIfaceName string, dhcpOptions *v1.DHCPOptions) error {
+	if !config.IPAMDisabled {
```

Comment:
Why was it returned back to here?
I believe having it in the `EnsureDhcpServerStarted` is more correct. `startDhcp` should just start the server, not checking of it needs to start it.